plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -544,14 +503,10 @@ func PodMatchesNodeLabels(pod *api.Pod, node *api.Node) bool {
 	return nodeAffinityMatches
 }
 
-type NodeSelector struct {
-	info NodeInfo
-}
-
-func (n *NodeStatus) PodSelectorMatches(pod *api.Pod, nodeName string, nodeInfo *schedulercache.NodeInfo) (bool, error) {
-	node, err := n.info.GetNodeInfo(nodeName)
-	if err != nil {
-		return false, err
+func PodSelectorMatches(pod *api.Pod, nodeName string, nodeInfo *schedulercache.NodeInfo) (bool, error) {
+	node := nodeInfo.Node()
+	if node == nil {
+		return false, fmt.Errorf("node not found: %q", nodeName)
```

Comment:
Nit: I am not sure about the k8s policy here. But it is not a good idea to return a very general but private defined error in a public function.